### ISSUE TITLE
Support for Lighthouse

### ISSUE THREAD

**Issue by kushalhalder** [thomasdondorf/puppeteer-cluster#348]
Essentially, lighthouse needs the port at which the chrome instance is running to run a job on the tab. Is that possible here?

**Comment by jasonreiche** [comment]
You can get the port by getting the browser from the page and then the power from the browser as demonstrated here:
```
flags.port = (new URL(page.browser().wsEndpoint())).port;
const result = await lighthouse(url, flags, config);
```
The problem I'm finding is that when Lighthouse takes over the Puppeteer instance it kills the puppeteer task before it has completed.  This results in Lighthouse not being about to find the browser.

**Comment by kushalhalder** [comment]
Yeah, the integration looked quiet cumbersome. So I just created two files:
1. chrome.js for launching chrome and getting the port
```
const puppeteer = require('puppeteer');
// const lighthouse = require('lighthouse');
const {URL} = require('url');

(async() => {
	const url = 'https://www.chromestatus.com/features';

	// Use Puppeteer to launch headful Chrome and don't use its default 800x600 viewport.
	const browser = await puppeteer.launch({
		headless: false,
		defaultViewport: null,
	});

	// // Wait for Lighthouse to open url, then inject our stylesheet.
	browser.on('targetchanged', async target => {
			const page = await target.page();
	//   if (page && page.url() === url) {
	//     await page.addStyleTag({content: '* {color: red}'});
	//   }
		console.log(page.url());
	});

	console.log((new URL(browser.wsEndpoint())).port)
})();
```
2. lighthouse.js for connecting to the chrome
```
const lighthouse = require('lighthouse');

var myArgs = process.argv.slice(2);
console.log('myArgs: ', myArgs);
(async() => {

	port = myArgs[0]
	url = myArgs[1]
	const {lhr} = await lighthouse(url, {
		port: port,
		output: 'json',
		logLevel: 'info',
	});
})();
```

**Comment by jasonreiche** [comment]
I found my issue was the concurrency model.
CONCURRENCY_CONTEXT ties the browser to the incognito window which gets lost during the handover to Lighthouse and closes the task.  The other models do work with varying results.

**Comment by kacbrz** [comment]
Hi @jasonreiche & @kushalhalder

Could you please describe in more detail how were you able to access wsEndpoint simultaneously running Puppeteer Cluster (not a single instance)? I tried both of your examples but I without any success.

**Comment by jasonreiche** [comment]
@kacbrz, I was not able to get multiple instances of Lighthouse to run concurrently using puppeteer-cluster.  Lighthouse is not designed to run concurrently as that would skew performance results.  Lighthouse conflicts with itself if multiple instances are running in the [same Node process](https://github.com/GoogleChrome/lighthouse/issues/7104#issuecomment-458368476).  You could look at using [lighthouse-batch-parallel](https://www.npmjs.com/package/lighthouse-batch-parallel).  When I need to run Lighthouse, I just use puppeteer-cluster to manage the pool and run a concurrency of 1.  In my case, I'm just looking for accessibility scores so we are switching over to @axe-core/webdriverjs moving forward.

**Comment by kacbrz** [comment]
@jasonreiche Thank you for replying! Last couple of days I was thinking about the performance in the exact same context you've provided. Despite that I still have to build a cluster of Lighthouses and somehow handle the performance issue. My small project just evolved and got a little bit more difficult...

**Comment by kushalhalder** [comment]
Hi @kacbrz, my use case was different. I did not have a cluster, and I also had only one machine. What I wanted was that I launch a chrome instance, which exposes a port through which I can push lighthouse events which run in different tabs as separate sessions. The above snippets simulate this.
